src/consensus/chainvalid.md

```diff
@@ -0,0 +1,116 @@
+# Chain validity
+
+A chain of blocks has to be valid in order to be considered for adoption. In
+particular the mainnet chain _is_ valid.
+
+Validity is defined by induction on the length of the chain as:
+- The first block on the chain fragment is valid,
+- The tail of the chain fragment is valid.
+
+For a block to be considered valid it has to be valid in three directions:
+
+- The envelope of the header must be valid, validity in this dimension is a
+  Consensus responsibility.
+- The header must be valid, validity in this dimension is defined by the
+  Consensus Protocol in effect.
+- The body must be valid, validity in this dimension is defined by the Ledger
+  era in effect. This check belongs to the Ledger layer so we will omit its
+  details in the rest of the document.
+
+From the description below, we omit the Ouroboros Classic case as such protocol
+has been effectively retired once (P)BFT were implemented. On the original
+implementation of Ouroboros Classic, the concept of Epoch Boundary Blocks (EBBs)
+was introduced, blocks which were made redundant by Ouroboros BFT. Such blocks
+had the peculiarity of sharing the slot number with the parent block and the
+block number with the subsequent block.
+
+<!-- toc -->
+
+## The Envelope of a header
+
+The Envelope of a header consists of the ledger-independent data of a block,
+such as block number, slot, hash, shape of the block, size, ... These checks are
+independent of the actual Consensus Protocol in effect and are used as a first
+sanity check on received blocks.
+
+An envelope is valid if:
```

Comment:
I was thinking, perhaps this document should be complemented with the different formal specs. Other aspects that could be linked (and maybe formally spec'd in the future) are the different chain selection rules described in this document.

Reply:
Sure, this can be improved later on!

Reply:
Added a couple of todos